R/cal-apply.R

```diff
@@ -0,0 +1,98 @@
+#---------------------------------- Apply --------------------------------------
+
+#' Applies a calibration to a set of prediction probabilities
+#' @details It currently supports data.frames only. It extracts the `truth` and
+#' the estimate columns names, and levels, from the calibration object.
+#' @param .data An object that can process a calibration object.
+#' @param calibration The calibration object (`cal_object`).
+#' @param ... Optional arguments; currently unused.
+#' @examples
+#' w_calibration <- cal_estimate_logistic(segment_logistic, Class)
+#'
+#' cal_apply(segment_logistic, w_calibration)
+#' @export
+cal_apply <- function(.data, calibration, ...) {
+  UseMethod("cal_apply")
+}
+
+#' @export
+cal_apply.data.frame <- function(.data, calibration, ...) {
+  if (calibration$type == "binary") {
+    cal_add_adjust(calibration, .data)
+  } else {
+    stop_multiclass()
+  }
+}
+
+# ------------------------------- Adjust ---------------------------------------
+
+cal_add_adjust <- function(calibration, .data) {
```

Comment:
What does this adjust?

Reply:
The probabilities, it's the method that wraps the the functions that actually make changes to the probability columns

Reply:
Sounds good. Maybe add a code comment to that effect